src/bindings/swig/lua/tests/CMakeLists.txt

```diff
@@ -25,7 +25,18 @@ macro (do_lua_test source)
 	endif ()
 endmacro (do_lua_test)
 
-file (GLOB TESTS *.lua)
-foreach (file ${TESTS})
-	do_lua_test (${file})
-endforeach (file ${TESTS})
+
+find_package (Lua 5.1)
+if (LUA_FOUND AND LUA_EXECUTABLE_VERSION_MATCHED)
+	file (GLOB TESTS *.lua)
+	foreach (file ${TESTS})
+		do_lua_test (${file})
+	endforeach (file ${TESTS})
+elseif(LUA_FOUND AND (NOT LUA_EXECUTABLE_VERSION_MATCHED))
+	message (WARNING "Lua interpreter does not match Lua libraries version, needed for SWIG Lua bindings tests. Tests will not be executed.")
+
+else ()
+	message (WARNING "Lua libraries not found, needed for SWIG Lua bindings tests. Tests will not be executed.")
+endif ()
+
+
```

Comment:
imho it would be better to move these lines over to `swig/lua/CMakeLists.txt` and enwrap the `add_subdirectory(tests)` call.
+ please remove the two trailing empty lines